models/classes/runner/QtiRunnerService.php

```diff
@@ -1156,13 +1082,62 @@ public function move(RunnerServiceContext $context, $direction, $scope, $ref)
         return $result;
     }
 
+    private function isSessionSuspended(RunnerServiceContext $context): bool
+    {
+        $session = $context->getTestSession();
+
+        if ($session instanceof AssessmentTestSession) {
+            if ($session->getState() == AssessmentTestSessionState::SUSPENDED) {
+                $this->getLogger()->debug(
+                    sprintf('%s DeliveryExecution is suspended', self::class)
+                );
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function isExecutionPaused(RunnerServiceContext $context): bool
+    {
+        $executionService = $this->getDeliveryExecutionService();
+        if ($executionService !== null) {
+            $execution = $executionService->getDeliveryExecution(
+                $context->getTestExecutionUri()
+            );
+
+            if ($execution->getState()->getUri() === DeliveryExecutionInterface::STATE_PAUSED) {
+                $this->getLogger()->debug(
+                    sprintf('%s DeliveryExecution is Paused', self::class)
+                );
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function getDeliveryExecutionService(): ?TaoDeliveryServiceProxy
+    {
+        if (
+            ($this->deliveryExecutionService === null)
+            && (class_exists(TaoDeliveryServiceProxy::class))
```

Comment:
As I know, there's no need to add external braces 🙂 
```suggestion
            $this->deliveryExecutionService === null
            && class_exists(TaoDeliveryServiceProxy::class)
```